periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
- adapt https://github.com/kubeflow/model-registry/pull/491
with 43cf0543b56716015059868aef220d178505c11b
- keep https://github.com/kubeflow/model-registry/pull/494
- keep https://github.com/kubeflow/model-registry/pull/500
- keep https://github.com/kubeflow/model-registry/pull/501
- keep https://github.com/kubeflow/model-registry/pull/502
- keep https://github.com/kubeflow/model-registry/pull/507
- keep https://github.com/kubeflow/model-registry/pull/506
- keep https://github.com/kubeflow/model-registry/pull/508
- adapting manifests with 43cf0543b56716015059868aef220d178505c11b for ODH

maintaining labels,

superseeds https://github.com/opendatahub-io/model-registry/pull/140

tarilabs commented on internal/converter/mlmd_converter_util_test.go:
```diff
@@ -44,6 +44,25 @@ func TestInt64ToString(t *testing.T) {
+	converted, err := StringToInt32(valid)
+	assertion.Nil(err)
+	assertion.Equal(int32(12345), converted)
+	assertion.Nil(StringToInt64(nil))
```
not sure about this line, seems is using the int64 variant... but not sure what was the intent

lampajr replied:
it was indeed a leftover, fixed :rocket:

tarilabs reviewed (approved):
minor comment, looks good to me, thanks!

lampajr commented on internal/converter/mlmd_converter_util_test.go:
```diff
@@ -44,6 +44,25 @@ func TestInt64ToString(t *testing.T) {
+	converted, err := StringToInt32(valid)
+	assertion.Nil(err)
+	assertion.Equal(int32(12345), converted)
+	assertion.Nil(StringToInt64(nil))
```
```suggestion
	assertion.Nil(StringToInt32(nil))
```

tarilabs commented on internal/converter/mlmd_converter_util_test.go:
```diff
@@ -44,6 +44,24 @@ func TestInt64ToString(t *testing.T) {
+	invalid := "not-a-number"
+	converted, err := StringToInt32(invalid)
+	assertion.NotNil(err)
+	assertion.Equal(int32(0), converted)
```
```suggestion

```

I'd also remove this line, as the Tuple_0 when error in Tuple_1 is present should be "unspecified" my2c

lampajr replied:
make sense!

lampajr replied:
had to ignore the function result using `_` :)

tarilabs reviewed (approved):
Thank you, sorry for spotting on the second round only

Al-Pragliola committed:
feat(csi): add CSI manifests (#491)

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

dependabot[bot] committed:
build(deps): bump github.com/kubeflow/model-registry from 0.2.8-alpha to 0.2.9 in /csi (#494)

Bumps [github.com/kubeflow/model-registry](https://github.com/kubeflow/model-registry) from 0.2.8-alpha to 0.2.9.
- [Release notes](https://github.com/kubeflow/model-registry/releases)
- [Commits](https://github.com/kubeflow/model-registry/compare/v0.2.8-alpha...v0.2.9)

---
updated-dependencies:
- dependency-name: github.com/kubeflow/model-registry
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump mypy from 1.11.2 to 1.12.1 in /clients/python (#500)

Bumps [mypy](https://github.com/python/mypy) from 1.11.2 to 1.12.1.
- [Changelog](https://github.com/python/mypy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/python/mypy/compare/v1.11.2...v1.12.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump ruff from 0.6.9 to 0.7.0 in /clients/python (#501)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.6.9 to 0.7.0.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.6.9...0.7.0)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump coverage from 7.6.3 to 7.6.4 in /clients/python (#502)

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.6.3 to 7.6.4.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.6.3...7.6.4)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump huggingface-hub from 0.25.2 to 0.26.1 in /clients/python (#507)

Bumps [huggingface-hub](https://github.com/huggingface/huggingface_hub) from 0.25.2 to 0.26.1.
- [Release notes](https://github.com/huggingface/huggingface_hub/releases)
- [Commits](https://github.com/huggingface/huggingface_hub/compare/v0.25.2...v0.26.1)

---
updated-dependencies:
- dependency-name: huggingface-hub
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tarilabs committed:
chore: format kustomization file (#506)

to avoid spurious formatting during relese
such as for example:
https://github.com/kubeflow/model-registry/pull/488/files

make the base formatting of this file consistent with
the result of `kustomize edit set image ...`

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

Al-Pragliola committed:
feat(csi): support multiple model registries (#508)

* feat(csi): support multiple model registries

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* chore(csi): add info about the uri in readme

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix(csi): remove unnecessary comment

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* chore(csi): cleanup after each test + improve log message after uri parsing

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* chore(csi): remove unnecessary kind external config in CI

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

---------

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

tarilabs committed:
Merge sync remote-tracking branch 'upstream/main' into tarilabs-20241028-sync

tarilabs committed:
k8s: align CSI manifest to ODH

adapts https://github.com/kubeflow/model-registry/pull/491

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
